files/build_templates/sonic_debian_extension.j2

```diff
@@ -401,6 +409,25 @@ sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS ta
 sudo LANG=C chroot $FILESYSTEM_ROOT docker $SONIC_NATIVE_DOCKERD_FOR_DOCKERFS tag {{imagename}}:latest {{imagebasename}}:latest
 {% endif %}
 {% endfor %}
+
+            
+{% if install_kubernetes == "y" %}
+## Pull in kubernetes docker images
+echo "pulling universal k8s images ..."
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT docker pull k8s.gcr.io/pause:3.2
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT docker pull k8s.gcr.io/kube-proxy:v1.18.0
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT docker pull calico/node:v3.12.0
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT docker pull calico/pod2daemon-flexvol:v3.12.0
+sudo https_proxy=$https_proxy LANG=C chroot $FILESYSTEM_ROOT docker pull calico/cni:v3.12.0
+echo "docker images pull complete"
+## Until we bootstrap scripts to bring up kubernetes
+sudo cp $IMAGE_CONFIGS/kubernetes/scripts/kube_join.sh $FILESYSTEM_ROOT/usr/bin/
+sudo cp $IMAGE_CONFIGS/kubernetes/scripts/kube_reset.sh $FILESYSTEM_ROOT/usr/bin/
+## Pod's hooks on PostStart & PreStop
+sudo cp $IMAGE_CONFIGS/kubernetes/scripts/pod_hook.sh $FILESYSTEM_ROOT_ETC_SONIC_PODS/
+sudo cp $IMAGE_CONFIGS/kubernetes/scripts/pod_readiness.sh $FILESYSTEM_ROOT_ETC_SONIC_PODS/
```

Comment:
I don't understand why we need these scripts and they seems premature.

If we really need it in future, we could submit another PR on a submodule, or in sonic-utilities repo. It should come with a design doc to explain the intentions and workflows.

We could narrow the scope of this PR to the packages and fundamental docker containers.

Reply:
All containers that are to run in SONiC, should be mandated to use pod_hook.sh for start/stop. These scripts record the containers presence in a single way across all. You can look at the scripts to review the behavior.

I am removing the pod_readiness.sh as that is expensive and would need some requirements to justify.